arch/xtensa/src/common/xtensa_swint.c

```diff
@@ -132,7 +132,7 @@ int xtensa_swint(int irq, void *context, void *arg)
       case SYS_restore_context:
         {
           DEBUGASSERT(regs[REG_A3] != 0);
-          CURRENT_REGS = (uint32_t *)regs[REG_A3];
+          CURRENT_REGS = *(uint32_t **)regs[REG_A3];
```

Comment:
don't need change

Reply:
This is similar to the discussion above.

Reply:
Ok, let's restore it after cp state move to stack.